qcodes/tests/test_station.py

```diff
@@ -659,6 +668,24 @@ def test_deprecated_limits_keyword_as_string():
       ch1:
         limits: -10, 10
     """)
-    with warnings.catch_warnings(record=True) as w:
+    with warnings.catch_warnings(record=True) as ws:
         st.load_instrument('mock')
-    assert issubclass(w[-1].category, QCoDeSDeprecationWarning)
+    assert any(issubclass(w.category, QCoDeSDeprecationWarning)
+               for w in ws)
+
+
+def test_config_validation_failure():
+    with pytest.raises(ValidationWarning):
+        station_from_config_str("""
+instruments:
+  mock:
+    driver: qcodes.tests.instrument_mocks.DummyInstrument
+invalid_keyword:
+  more_errors: 42
+    """)
+
+
+def test_config_validation_comprehensive_config():
+    Station(config_file=os.path.join(
```

Comment:
interesting that there are no traces of QCoDeSDeprecationWarning in the stdout of running pytest coming from the fact that the yaml file contains driver+type entries and parameter limits as a string. i expect those to show up because other warnings to pop up, [for example this](https://dev.azure.com/ms-quantum-public/Microsoft%20Quantum%20(public)/_build/results?buildId=4929&view=logs&j=d84cbf5c-d59c-5b28-7c9f-55b9660cb0fb&t=97624319-fa93-5383-4c5b-1ee5f22c3215&l=145). @jenshnielsen , is it all again due to the funky default behaviour of warnings in python?

to clarify, my expectation was that this call would be wrapped in smth like "ignore warnings".

Reply:
Pytest does not show all warnings by default. Here we are raising a `QCoDeSDeprecationWarning` which is a `RuntimeWarning`. It does not get shown by default. We should consider making these warnings errors by adding the `-W` parameter or changing the pytest.ini 
```
[pytest]
filterwarnings =
    error::QCoDeSDeprecationWarning
```
(see here https://docs.pytest.org/en/latest/warnings.html)
Then we have to add
```
@pytest.mark.filterwarnings("default::QCoDeSDeprecationWarning")
```
to the tests that expect warnings.